src/gluonts/dataset/loader.py

```diff
@@ -37,6 +50,63 @@ def __call__(self, data, is_train):
         yield from batcher(data, self.batch_size)
 
 
+class Stack(Transformation, BaseModel):
+    def __call__(self, data, is_train):
+        for batch in data:
+            yield rows_to_columns(batch, np.array)
+
+
+def as_stacked_batches(
+    dataset: Dataset,
+    *,
+    batch_size: int,
+    output_type: Callable,
+    num_batches_per_epoch: Optional[int] = None,
+    shuffle_buffer_length: Optional[int] = None,
+    field_names: Optional[list] = None,
+):
+    """
+    Prepare data in batches to be passed to a network.
+
+    Input data is collected into batches of size ``batch_size`` and then
+    columns are stacked on top of each other. In addition, the result is
+    wrapped in ``output_type``. This is needed since our pipelines generally
+    use ``numpy`` arrays, but networks expect framework specific data formats
+    (i.e. ``torch.tensor`` and ``mxnet.ndarray``).
+
+    If ``num_batches_per_epoch`` is provided, only those number of batches are
+    effectively returned. This is especially useful for training when
+    providing a cyclic dataset.
+
+    To pseudo shuffle data, ``shuffle_buffer_length`` can be set to collect
+    inputs into a buffer first, from which we then randomly sample.
+
+    Setting ``field_names`` will only consider those columns in the input data
+    and discard all other values.
+    """
+
+    if shuffle_buffer_length:
+        dataset = PseudoShuffled(dataset, shuffle_buffer_length)
+
+    transform: Transformation = Identity()
+
+    if field_names is not None:
+        transform += SelectFields(field_names)
+
+    transform += Batch(batch_size=batch_size)
+    transform += Stack()
```

Comment:
This could also be an optional parameter, and only stack if required.

On the other hand, I don't know where we would want that.

Reply:
yeah no need for that